src/zoo/owl_zoo_dir.ml

```diff
@@ -3,55 +3,101 @@
  * Copyright (c) 2016-2018 Liang Wang <liang.wang@cl.cam.ac.uk>
  *)
 
+let dir = Owl_zoo_log.dir
 
-let rec _extract_zoo_gist f added =
+let parse_gist_string gist =
+  let strip_string s =
+    Str.global_replace (Str.regexp "[\r\n\t ]") "" s
+  in
+  let validate_len s len info =
+    if ((String.length s) = len) then s
+    else failwith ("Zoo error: invalid " ^ info ^ ":" ^ s)
+  in
+  let regex = Str.regexp "/" in
+  let lst = Str.split_delim regex gist
+    |> List.map strip_string in
+  let gid = List.hd lst in
+  let len = List.length lst in
+  let vid =
+    if (len = 1 ) then Owl_zoo_log.find_latest_vid_remote gid
+    else if (len = 2 || len = 3) then (
+      let v = List.nth lst 1 in
+      match v with
+      | "remote" -> Owl_zoo_log.find_latest_vid_remote gid
+      | "latest" -> Owl_zoo_log.find_latest_vid_local  gid
+      | _        -> v
+    )
+    else failwith "Zoo error: Illegal parameters numbers"
+  in
+  let pin =
+    if (len = 1 || len = 2) then false
+    else if (len = 3) then (
+      if (List.nth lst 2 = "pin") then true else false
+    )
+    else failwith "Zoo error: Illegal parameters numbers"
+  in
+  (validate_len gid 32 "gist id"), (validate_len vid 40 "version id"), pin
+
+
+let _deploy_gist gid vid =
```

Comment:
A bit confused by the name, so this deploy is just download/install zoo gist locally, it has nothing to do with the zoo depolyer we discussed before, right? If so, name probably needs to be changed to avoid ambiguity.

Reply:
Yes, this is just a ambiguous name off the top of my head. I'll change it.